src/pages/workspaces/workspace/useDeleteWorkspaceState.ts

```diff
@@ -0,0 +1,203 @@
+import _ from 'lodash/fp'
+import { useRef, useState } from 'react'
+import { Ajax } from 'src/libs/ajax'
+import { withErrorReportingInModal } from 'src/libs/error'
+import { useCancellation, useOnMount } from 'src/libs/react-utils'
+import { getUser } from 'src/libs/state'
+import * as Utils from 'src/libs/utils'
+import { BaseWorkspace, isAzureWorkspace, isGoogleWorkspace, WorkspaceInfo } from 'src/libs/workspace-utils'
+import { isResourceDeletable } from 'src/pages/workspaces/workspace/analysis/runtime-utils'
+
+
+export const WorkspaceResourceDeletionPollRate = 5000
+
+interface DeleteWorkspaceModalLeoApp {
```

Comment:
FYI IA, I'd like to pull these into real models in the Apps + Runtimes Ajax space at some point, not sure if you all have any plans about typescriptifying that area.

Reply:
I'm fairly certain we do have plans for that.
Apps and Runtimes touch a lot of code, so I think we were planning to wait until we get more smaller pieces converted to .ts before tackling this.